Okay, it's not that easy. This unfortunately completely breaks several node types (like workers):

```
org.apache.felix.resolver.reason.ReasonException: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=opencast-worker; type=karaf.feature; version="[10.0.0.SNAPSHOT,10.0.0.SNAPSHOT]"; filter:="(&(osgi.identity=opencast-worker)(type=karaf.feature)(version>=10.0.0.SNAPSHOT)(version<=10.0.0.SNAPSHOT))" [caused by: Unable to resolve opencast-worker/10.0.0.SNAPSHOT: missing requirement [opencast-worker/10.0.0.SNAPSHOT] osgi.identity; osgi.identity=opencast-core; type=karaf.feature [caused by: Unable to resolve opencast-core/10.0.0.SNAPSHOT: missing requirement [opencast-core/10.0.0.SNAPSHOT] osgi.identity; osgi.identity=opencast-metrics-exporter; type=osgi.bundle; version="[10.0.0.SNAPSHOT,10.0.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve opencast-metrics-exporter/10.0.0.SNAPSHOT: missing requirement [opencast-metrics-exporter/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.assetmanager.api.AssetManager)"; effective:=active]]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        at org.apache.felix.resolver.ResolverImpl.doResolve(ResolverImpl.java:392)
        at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:378)
        at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:332)
        at org.apache.karaf.features.internal.region.SubsystemResolver.resolve(SubsystemResolver.java:257)
        at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:393)
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1062)
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$13(FeaturesServiceImpl.java:998)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-worker/10.0.0.SNAPSHOT: missing requirement [opencast-worker/10.0.0.SNAPSHOT] osgi.identity; osgi.identity=opencast-core; type=karaf.feature [caused by: Unable to resolve opencast-core/10.0.0.SNAPSHOT: missing requirement [opencast-core/10.0.0.SNAPSHOT] osgi.identity; osgi.identity=opencast-metrics-exporter; type=osgi.bundle; version="[10.0.0.SNAPSHOT,10.0.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve opencast-metrics-exporter/10.0.0.SNAPSHOT: missing requirement [opencast-metrics-exporter/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.assetmanager.api.AssetManager)"; effective:=active]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        ... 12 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-core/10.0.0.SNAPSHOT: missing requirement [opencast-core/10.0.0.SNAPSHOT] osgi.identity; osgi.identity=opencast-metrics-exporter; type=osgi.bundle; version="[10.0.0.SNAPSHOT,10.0.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve opencast-metrics-exporter/10.0.0.SNAPSHOT: missing requirement [opencast-metrics-exporter/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.assetmanager.api.AssetManager)"; effective:=active]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        ... 13 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-metrics-exporter/10.0.0.SNAPSHOT: missing requirement [opencast-metrics-exporter/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.assetmanager.api.AssetManager)"; effective:=active
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        ... 14 more
```

The reason for that is the static OSGi binding to the asset manager which does not exist on those nodes.
Three possible ways to go forward:

- Leave it on admin-like nodes only for now, adjusting the documentation to not suggest that it will work on other nodes
- Turn the static into a dynamic binding and exclude the metrics which are not available.
- Run different implementations on admin-like nodes and other nodes

--- a/modules/metrics-exporter/src/main/java/org/opencastproject/metrics/impl/MetricsExporter.java
+++ b/modules/metrics-exporter/src/main/java/org/opencastproject/metrics/impl/MetricsExporter.java
@@ -38,6 +38,8 @@ import org.osgi.framework.Version;
 import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -206,10 +208,12 @@ public class MetricsExporter {
     }
 
     // Get numbers from asset manager
-    for (Organization organization: organizationDirectoryService.getOrganizations()) {
-      eventsInAssetManager
-          .labels(organization.getId())
-          .set(assetManager.countEvents(organization.getId()));
+    if (assetManager != null) {
+      for (Organization organization: organizationDirectoryService.getOrganizations()) {
+        eventsInAssetManager
+            .labels(organization.getId())
+            .set(assetManager.countEvents(organization.getId()));
+      }
     }
 
     // collect metrics
@@ -228,8 +232,17 @@ public class MetricsExporter {
     this.organizationDirectoryService = organizationDirectoryService;
   }
 
-  @Reference
+  @Reference(
+      policy = ReferencePolicy.DYNAMIC,
+      cardinality = ReferenceCardinality.OPTIONAL,
+      unbind = "unsetAssetManager"
+  )
   public void setAssetManager(AssetManager assetManager) {
     this.assetManager = assetManager;
   }
+
+  public void unsetAssetManager(AssetManager assetManager) {
+    this.assetManager = null;
+  }
+
 }

--- a/modules/metrics-exporter/src/main/java/org/opencastproject/metrics/impl/MetricsExporter.java
+++ b/modules/metrics-exporter/src/main/java/org/opencastproject/metrics/impl/MetricsExporter.java
@@ -125,11 +125,7 @@ public class MetricsExporter {
       .help("Version of Opencast (based on metrics module)")
       .labelNames("part")
       .register();
-  private final Gauge eventsInAssetManager = Gauge.build()
-      .name("opencast_asset_manager_events")
-      .help("Events in Asset Manager")
-      .labelNames("organization")
-      .register();
+  private Gauge eventsInAssetManager;
 
   /** OSGi services */
   private ServiceRegistry serviceRegistry;
@@ -239,10 +235,16 @@ public class MetricsExporter {
   )
   public void setAssetManager(AssetManager assetManager) {
     this.assetManager = assetManager;
+    eventsInAssetManager = Gauge.build()
+        .name("opencast_asset_manager_events")
+        .help("Events in Asset Manager")
+        .labelNames("organization")
+        .register();
   }
 
   public void unsetAssetManager(AssetManager assetManager) {
     this.assetManager = null;
+    registry.unregister(eventsInAssetManager);
   }
 
 }

